strawberryfields/backends/gaussianbackend/backend.py

```diff
@@ -154,6 +173,22 @@ def loss(self, T, mode):
     def thermal_loss(self, T, nbar, mode):
         self.circuit.thermal_loss(T, nbar, mode)
 
+    def measure_fock(self, modes, shots=1, select=None):
+        if select is not None:
+            raise NotImplemented("Postselection is currently not supported "
+                                 "with measure_fock on the gaussian backend")
+
+        # TODO: add logic to make sure the correct modes are measured
+        mu = self.circuit.mean
+        cov = self.circuit.scovmatxp()
+        # check we are sampling from a gaussian state with zero mean
+        if not allclose(mu, zeros_like(mu)):
+            raise NotImplementedError("PNR measurement is only supported for "
+                                      "Gaussian states with zero mean")
```

Comment:
@nquesada, if the sampling is done with loop hafnians, then can displacements be taken into account?

Reply:
> Yes, however this should be handled by the Hafnian library not by sf.

True, we should add an option to the sampling function to accept displacements